### PR TITLE
Viruses with only positive symptoms have a special HUD icon, full heals don't remove positive viruses.

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
add: Positive viruses are now recognized by medical HUDs with a special icon.
tweak: fully_heal no longer removes nonharmful viruses. This affects legion souls, staffs of healing, and changeling's revive.
/:cl:

Previously it just showed people as healthy, like a stealth virus.

The new icon: 
hudbuff

### CODE REVIEW

**Comment by kevinz000** [comment]
While you're at it make admin fullheals not impact positive viruses, please.

File: code/game/data_huds.dm [hunk 1]
```diff
@@ -62,8 +62,11 @@
 //called when a carbon changes virus
 /mob/living/carbon/proc/check_virus()
 	for(var/datum/disease/D in viruses)
-		if((!(D.visibility_flags & HIDDEN_SCANNER)) && (D.severity != NONTHREAT))
-			return 1
+		if(!(D.visibility_flags & HIDDEN_SCANNER))
+			if (D.severity == NONTHREAT) //a buffing virus gets an icon
+				return 2
```
Comment (phil235) [inline code/game/data_huds.dm]: so if I have a healing virus AND a bad virus, the med hud still shows a smile?

File: code/game/data_huds.dm [hunk 2]
```diff
@@ -160,8 +163,10 @@
 		holder.icon_state = "hudxeno"
 	else if(stat == DEAD || (status_flags & FAKEDEATH))
 		holder.icon_state = "huddead"
-	else if(check_virus())
+	else if(check_virus() == 1)
 		holder.icon_state = "hudill"
+	else if(check_virus() == 2)
+		holder.icon_state = "hudbuff"
```
Comment (phil235) [inline code/game/data_huds.dm]: maybe we can change the code so check_virus() is only called once.

File: code/game/data_huds.dm (hunk 1 above)
Comment (kevinz000) [inline code/game/data_huds.dm]: emergent gameplay

**Comment by XDTM** [comment]
Now it properly prioritizes harmful viruses, and check_virus() is only called once. I also added kevinz000's suggestion.

**Comment by ma44** [comment]
What if you have something like sneezing/coughing to make the virus spread

**Comment by XDTM** [comment]
Then it still counts as bad, just like it does now

**Review by phil235 — APPROVED** [review]
:+1:

**Comment by ExcessiveUseOfCobblestone** [comment]
Sneezing is the worst symptom [sans coughing, which alone should warrant a lynch].

**Comment by Ergovisavi** [comment]
I'm going to :-1: this. Not because the code is bad, or the thought that positive virii not showing the player as sick is a bad thing, but because I don't believe virology is good for the game as a whole and anything we do to more ingrain it into the system is a bad thing.

File: code/game/data_huds.dm (hunk 1 above)
Comment (XDTM) [inline code/game/data_huds.dm]: bad virus icon has a priority now

**Comment by Shapsy** [comment]
Being able to trust a virus 100% without scanning it is lame

**Comment by XDTM** [comment]
It's nothing that you couldn't do before, Shaps, since it would appear invisible on the HUD but visible on scanners.

**Comment by InsaneHyena** [comment]
> virology buffs

No, thanks.

**Comment by AdamElTablawy** [comment]
what if someone who had a healing virus look like they had a xeno egg in them on the health hud so angry doctors would beat them to crit and fruitlessly do surgery on them for 10 minutes for having a heal virus

yes? yes????

**Comment by XDTM** [comment]
If you have a xeno egg it'll show up on the HUD first, otherwise it'll show the heal virus. Why would a doctor do surgery on someone that doesn't show the infestation icon?

**Comment by nullbear** [comment]
he's saying to replace the heal virus icon with the xeno egg icon, even if there is no xenoegg. It's a troll.

**Comment by XDTM** [comment]
Oh.

**Pull request merged**